pkg/certificate/signing_requestor.go

```diff
@@ -215,40 +241,64 @@ func (s *signingRequestorImpl) Issue(ctx context.Context, name string, ips []str
 		Obj:    resource.MustToUnstructured(newSecret),
 		MutateOnUpdate: func(obj *unstructured.Unstructured) (*unstructured.Unstructured, error) {
 			existing := resource.MustFromUnstructured(obj, &corev1.Secret{})
-			if !bytes.Equal(existing.Data[CSRDataKey], newSecret.Data[CSRDataKey]) {
-				// The CSR data changed so clear the signed annotation, so it will be re-signed.
-				delete(existing.Annotations, RequestSignedLabelKey)
+
+			// Check if we need to regenerate the CSR by comparing the IPs
+			// We need to extract the IPs from the existing CSR to compare
+			var needsNewCSR bool
+
+			// Parse existing CSR to extract IPs and compare
+			existingIPs, err := s.extractIPsFromCSR(existing.Data[CSRDataKey])
+			if err != nil {
+				logger.Warningf("Failed to parse existing CSR for secret %q: %v", existing.Name, err)
+				needsNewCSR = true
+			} else {
+				// Compare IP lists
+				needsNewCSR = !slices.Equivalent(existingIPs, ips, slices.Key[string])
 			}
 
-			existing.Data[PrivateKeyDataKey] = newSecret.Data[PrivateKeyDataKey]
-			existing.Data[CSRDataKey] = newSecret.Data[CSRDataKey]
+			if needsNewCSR {
+				existing.Data[PrivateKeyDataKey] = newSecret.Data[PrivateKeyDataKey]
+				existing.Data[CSRDataKey] = newSecret.Data[CSRDataKey]
+
+				// Clear signed annotation since we have new CSR data
+				delete(existing.Annotations, RequestSignedLabelKey)
+			}
+			// If needsNewCSR is false, we preserve the existing CSR data and signed state
 
 			return resource.MustToUnstructured(existing), nil
 		},
 	})
 	if err == nil {
-		s.onSignedMap.Store(newSecret.Name, onSigned)
+		// Track certificate for renewal (expiration will be set when certificate is signed)
+		s.issuedCerts.Store(newSecret, certInfo{
+			name:      name,
+			ips:       ips,
+			onSigned:  onSigned,
+			expiresAt: time.Time{}, // Will be updated when certificate is signed
+		})
 	}
 
 	if result == util.OperationResultCreated {
-		logger.Infof("Successfully created CSR Secret %q", newSecret.Name)
+		logger.Infof("Successfully created CSR Secret %q", newSecret)
```

Comment:
This dumps the whole secret which is quite large with the byte data.

```suggestion
		logger.Infof("Successfully created CSR Secret %q", newSecret.Name)
```